lib/Interpreter/Value.cpp

```diff
@@ -231,9 +231,15 @@ namespace cling {
   namespace valuePrinterInternal {
     std::string printTypeInternal(const Value& V);
     std::string printValueInternal(const Value& V);
+    extern const char* const kUndefined;
   } // end namespace valuePrinterInternal
 
   void Value::print(llvm::raw_ostream& Out, bool Escape) const {
+    if (!m_Interpreter) {
+      Out << valuePrinterInternal::kUndefined << "\n";
```

Comment:
> That make parsing cling's output and detecting invalid results, way more difficult.

Correct, but don't care: I'm after end user usability, not test simplicity. And yes, diagnostics help, but I don't see why they are a motivation to shove all differences in error versus invalid `Value` under the same rug?